src/wh_server_crypto.c

```diff
@@ -782,6 +798,10 @@ static int hsmCryptoCmac(whServerContext* server, whPacket* packet,
                 keyId = WH_MAKE_KEYID(WH_KEYTYPE_CRYPTO,
                     server->comm->client_id, packet->cmacReq.keyId);
             }
+            /* evict the aes sized key in the normal cache */
+            if (moveToBigCache == 1) {
+                ret = hsmEvictKey(server, keyId);
+            }
```

Comment:
so it seems that CMAC support now hinges on `WOLFHSM_CFG_SERVER_KEYCACHE_BIG_COUNT` being > 0, since the CMAC struct is cached across invocations. Since WOLFHSM_CFG_SERVER_KEYCACHE_BIG_COUNT is configurable by the user, and is very much allowed to be 0, I think we probably need to implement some sort of compile time checking for this? Otherwise it will (best case) result in potentially hard to diagnose runtime errors

Reply:
as long as the normal cache is big enough to hold the cmac struct it will work fine since `hsmEvictKey` and `hsmCacheKey` transparently put the key into the correct cache. if `WOLFHSM_CFG_SERVER_KEYCACHE_BIG_COUNT` is 0 and `WOLFHSM_CFG_SERVER_KEYCACHE_BUFSIZE` is big enough to hold a cmac struct it will work fine, if it's not big enough `WH_ERROR_NOSPACE` will be returned